### PR TITLE
Update CI to CUDA 11.7.1

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

we can also consider bumping to 11.7.1

Closes #16193 

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @akihironitta

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: dockers/base-cuda/Dockerfile [hunk 1]
```diff
@@ -13,13 +13,14 @@
 # limitations under the License.
 
 ARG UBUNTU_VERSION=20.04
-ARG CUDA_VERSION=11.6.1
+ARG CUDA_VERSION=11.7.1
```
Comment (carmocca) [inline dockers/base-cuda/Dockerfile:16]: This PR should also update the version used in gpu ci. Otherwise we arent sure if the image is working well

Comment (Borda) [reply dockers/base-cuda/Dockerfile:16]: well, some integration like collesalai has not been built yet for this CUDA version :otter:

Comment (carmocca) [reply dockers/base-cuda/Dockerfile:16]: But we already have two separate GPU ci jobs. You just need to update the latest one

Comment (carmocca) [reply dockers/base-cuda/Dockerfile:16]: This: https://github.com/Lightning-AI/lightning/blob/acdb1453f1a85441083f9b20e9d150cd6e9e8fd9/.azure/gpu-tests-pytorch.yml#L50

Also for Lite

Comment (Borda) [reply dockers/base-cuda/Dockerfile:16]: well, it seems that colossalAI is not built for PT 1.13 anyway
https://www.colossalai.org/download/

Comment (carmocca) [reply dockers/base-cuda/Dockerfile:16]: But `colossalai` is not installed in the "PyTorch - latest" job anyways. Only the `PyTorch & strategies`

Comment (Borda) [reply dockers/base-cuda/Dockerfile:16]: true :)

**Comment by Borda** [comment]
seems we need to fist update CUDA on our physical/HW hosts ideally to 11.8 or 12.0

**Comment by Borda** [comment]
Agter runninh on a machine with updated CUDA to 11.8 I see the following error:
- testing PyTorch - latest
	```
	=========================== short test summary info ============================
	FAILED models/test_horovod.py::test_horovod_multi_gpu - assert 139 == 0
	FAILED models/test_horovod.py::test_horovod_multi_gpu_accumulate_grad_batches - assert 139 == 0
	FAILED models/test_horovod.py::test_horovod_multi_gpu_grad_by_value - assert 139 == 0
	FAILED models/test_horovod.py::test_horovod_amp - assert 139 == 0
	FAILED models/test_horovod.py::test_horovod_gather - assert 139 == 0
	==== 5 failed, 165 passed, 108 skipped, 3495 warnings in 403.27s (0:06:43) =====
	```
	NOTE: Horovod shall not sun here as it is a strategy
- testing PyTorch & strategies
	```
	=========================== short test summary info ============================
	FAILED strategies/test_bagua_strategy.py::test_manual_optimization - AssertionError: Unexpected rsp=<Response [500]>
	==== 1 failed, 177 passed, 100 skipped, 3741 warnings in 443.73s (0:07:23) =====
	```

**Comment by Borda** [comment]
@carmocca any suggestion about this error:
```
_________________________ test_fsdp_rewrap_limitation __________________________

tmpdir = local('/tmp/pytest-of-root/pytest-50/test_fsdp_rewrap_limitation0')

    @RunIf(min_cuda_gpus=1, skip_windows=True, standalone=True, fairscale_fully_sharded=True)
    def test_fsdp_rewrap_limitation(tmpdir):
        trainer = Trainer(
            default_root_dir=tmpdir,
            accelerator="gpu",
            devices=1,
            max_steps=1,
            limit_val_batches=0,
            limit_test_batches=1,
            strategy="fsdp",
        )
        model = TestFSDPModelAutoWrapped()
>       trainer.fit(model)

strategies/test_ddp_fully_sharded_with_full_state_dict.py:253: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../src/pytorch_lightning/trainer/trainer.py:602: in fit
    call._call_and_handle_interrupt(
../../src/pytorch_lightning/trainer/call.py:36: in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
../../src/pytorch_lightning/strategies/launchers/subprocess_script.py:88: in launch
    return function(*args, **kwargs)
../../src/pytorch_lightning/trainer/trainer.py:644: in _fit_impl
    self._run(model, ckpt_path=self.ckpt_path)
../../src/pytorch_lightning/trainer/trainer.py:1044: in _run
    self._call_configure_sharded_model()  # allow user to setup in model sharded environment
../../src/pytorch_lightning/trainer/trainer.py:1290: in _call_configure_sharded_model
    with self.strategy.model_sharded_context():
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pytorch_lightning.strategies.fully_sharded.DDPFullyShardedStrategy object at 0x7f4808b6c610>

    @contextlib.contextmanager
    def model_sharded_context(self) -> Generator:
        log.detail(f"{self.__class__.__name__}: entered model_sharded_context.")
        precision = self.precision_plugin.precision
    
        def wrap_policy(*args: Any, **kwargs: Any) -> Any:
            return default_auto_wrap_policy(*args, **kwargs, min_num_params=self.min_num_params)
    
>       with enable_wrap(
            wrapper_cls=FullyShardedDataParallel,
            auto_wrap_policy=wrap_policy,
            process_group=self.process_group,
            cpu_offload=self.cpu_offload,
            move_grads_to_cpu=self.move_grads_to_cpu,
            flatten_parameters=self.flatten_parameters,
            mixed_precision=(precision in (PrecisionType.MIXED, PrecisionType.HALF)),
            reshard_after_forward=self.reshard_after_forward,
            fp32_reduce_scatter=self.fp32_reduce_scatter,
            compute_dtype=self.compute_dtype,
            bucket_cap_mb=self.bucket_cap_mb,
            state_dict_device=self.state_dict_device,
        ):
E       NameError: name 'enable_wrap' is not defined
```

**Comment by carmocca** [comment]
Fixed

**Review by carmocca — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]

**Pull request merged**